packages/ember-template-compiler/lib/plugins/assert-if-helper-without-arguments.js

```diff
@@ -0,0 +1,43 @@
+import { assert } from 'ember-debug';
+import calculateLocationDisplay from '../system/calculate-location-display';
+
+export default function assertIfHelperWithoutArguments(env) {
+  let { moduleName } = env.meta;
+
+  return {
+    name: 'assert-if-helper-without-arguments',
+
+    visitor: {
+      BlockStatement(node) {
+        if (isInvalidBlockIf(node)) {
+          assert(`#${assertMessage(node.path.original)} ${calculateLocationDisplay(moduleName, node.loc)}`);
+        }
+      },
+
+      MustacheStatement(node) {
+        if (isInvalidInlineIf(node)) {
+          assert(`#${assertMessage(node.path.original)} ${calculateLocationDisplay(moduleName, node.loc)}`);
+        }
+      },
+
+      SubExpression(node) {
+        if (isInvalidInlineIf(node)) {
+          assert(`#${assertMessage(node.path.original)} ${calculateLocationDisplay(moduleName, node.loc)}`);
+        }
+      }
+    }
+  };
+}
+
+function assertMessage(original) {
+  return `${original} requires a single argument.`;
+}
+
+function isInvalidInlineIf(node) {
+  return node.path.original === 'if' && (!node.params || node.params.length < 1);
```

Comment:
Should we make the minimum length be 2?  I'm not sure how something like `{{if something}}` is useful (e.g. if there is no truthy or falsey argument, what is even happening?)...

What do you think @AlexTraher?

Reply:
@rwjblue Yeah agreed - I like the idea it fails loudly.